pumpkin/src/client/mod.rs

```diff
@@ -503,7 +503,10 @@ impl Client {
                     return true;
                 }
                 Ok(None) => (), //log::debug!("Waiting for more data to complete packet..."),
-                Err(err) => log::warn!("Failed to decode packet for: {}", err.to_string()),
+                Err(err) => {
+                    log::warn!("Failed to decode packet for: {}", err.to_string());
+                    return true; // return to avoid reserving additional bytes
```

Comment:
We should drop if the packet is incorrect 
```suggestion
                    return false; // return to avoid reserving additional bytes
```

Reply:
Maybe even close the connection. Returning false just doesn't process packets without closing the connection

Reply:
Ok good idea on that, now implemented thanks !